pkg/services/accesscontrol/pluginutils/utils.go

```diff
@@ -60,6 +60,22 @@ func ToRegistrations(pluginID, pluginName string, regs []plugins.RoleRegistratio
 	return res
 }
 
+// PluginIDFromName extracts the plugin ID from the role name
+func PluginIDFromName(roleName string) string {
+	if !strings.HasPrefix(roleName, ac.PluginRolePrefix) {
+		return ""
+	}
+
+	pluginID := strings.Builder{}
+	for _, c := range roleName[len(ac.PluginRolePrefix):] {
+		if c == ':' {
+			break
+		}
+		pluginID.WriteRune(c)
+	}
```

Comment:
nit: could just split on `:` and take the second element. Would be slightly more readable to me, but same result.